lib/scidata/squad.ex

```diff
@@ -0,0 +1,254 @@
+
+      iex> Scidata.Squad.download()
+      [
+        %{
+          "paragraphs" => [
+            %{
+              "context" => "Architecturally, the school has a...",
+              "qas" => [
+                %{
+                  "answers" => [%{"answer_start" => 515, "text" => "Saint Bernadette Soubirous"}],
+                  "id" => "5733be284776f41900661182",
+                  "question" => "To whom did the..."
+                }, ...
+              ]
+            }
+          ],
+          "title" => "University_of_Notre_Dame"
+        }, ...
+      ]
+  """
+
+  def download() do
+    download_dataset(@train_dataset_file)
+  end
+
+  @doc """
+  Downloads the SQuAD test dataset
+
+  ## Examples
+
+      iex> Scidata.Squad.download_test()
+      [
+        %{
+          "paragraphs" => [
+            %{
+              "context" => "Super Bowl 50 was an American football game t...",
+              "qas" => [
+                %{
+                  "answers" => [
+                    %{"answer_start" => 177, "text" => "Denver Broncos"},...
+                  ],
+                  "id" => "56be4db0acb8001400a502ec",
+                  "question" => "Which NFL team represented the AFC at Super Bowl 50?"
+                },
+              ]
+            }
+          ],
+          "title" => "Super_Bowl_50"
+        }, ...
+      ]
+  """
+
+  def download_test() do
+    download_dataset(@test_dataset_file)
+  end
+
+  defp download_dataset(dataset_name) do
+    content =
+      Utils.get!(@base_url <> dataset_name).body
+      |> Jason.decode!()
+
+    content["data"]
+  end
+
+  @doc """
+  Convert a list of maps into a single map with containing list-type values.
+
+  ## Examples
+
+      iex> Scidata.Squad.flatten(entries)
+      %{
+        "answer_start" => [515, ...],
+        "context" => ["Architecturally, the...", ...],
+        "id" => ["5733be284776f41900661182", ...],
+        "question" => ["To whom did the Vir...", ...],
+        "text" => ["Saint Bernadette Soubirous", ...],
+        "title" => ["University_of_Notre_Dame", ...]
+      }
+
+  """
+
+  def flatten(entries) do
+    normalized_data =
+      for %{"paragraphs" => paragraph, "title" => title} <- entries,
+          %{"context" => context, "qas" => qas} <- paragraph,
+          %{"id" => id, "question" => question, "answers" => answers} <- qas,
+          answer <- answers do
+        Map.merge(answer, %{
+          "id" => id,
+          "context" => context,
+          "question" => question,
+          "title" => title
+        })
+      end
+
+    table = %{
+      "answer_start" => [],
+      "context" => [],
+      "id" => [],
+      "question" => [],
+      "text" => [],
+      "title" => []
+    }
+
+    normalized_data
+    |> Enum.reduce(table, fn map, acc ->
+      Enum.map(acc, fn {key, values} ->
+        {key, [map[key] | values]}
+      end)
+    end)
+    |> Enum.map(fn {key, values} -> {key, :lists.reverse(values)} end)
+    |> Enum.into(%{})
```

Comment:
I think we might not need this. What if we DF.from_list (or similar) instead of DF.from_map? They should have an API that accepts lists of rows?

Reply:
I see just `from_map/1` on [the Explorer docs](https://elixir-nx.github.io/explorer/Explorer.DataFrame.html#from_map/2) right now.

We can make the whole map inside the `for` loop though: https://gist.github.com/t-rutten/17cd76074ecdd3d6062c631a8a749963

Reply:
@t-rutten nice!

So final suggestion: let's go with @t-rutten code above, call it `to_map` (instead of `flatten`), remove the existing `to_maps`, and ship it?

I am so sorry for all the back and forth in this. :D

Reply:
Oh, what if we call it `to_columns` instead of `to_map`? It will make it clear we are returning the columns.

Reply:
Perfect :)

Reply:
I kind of like `to_map` so it's `Scidata.to_map` -> `Explorer.from_map`, but `to_columns` totally works. Do you have a preference @jeantux?

Reply:
`to_columns` sounds good to me